lib/evaluate.js

```diff
@@ -368,6 +394,22 @@ export const runFraudDetection = async (roundIndex, measurements, sparkRoundDeta
     }
   }
 
+  //
+  // 3. Group measurements to per-task committees and find majority result
+  //
+  const iterateHonestMeasurements = function * () {
+    for (const m of measurements) {
+      if (m.fraudAssessment !== 'OK') continue
+      yield m
+    }
+  }
+
+  const committees = groupMeasurementsToCommittees(iterateHonestMeasurements())
```

Comment:
Could this be simpler like this?

```suggestion
  const committees = groupMeasurementsToCommittees(
    measurements.filter(m => m.fraudAssessment === 'OK')
  )
```

Reply:
I didn't want to duplicate the array of measurements because there are hundreds of thousands of them. Do you think it's a preliminary optimisation to avoid now?

Reply:
If you didn't measure it then yes I think so

Reply:
I did a small test:

```js
{ const a = []; for (let i = 0; i < 1_000_000; i++) a.push({ beep: 'boop' }); const s = new Date(); [...a]; console.log(new Date() - s) }
52
```

52 milliseconds isn't little, especially with a way smaller object size. I'm +1 for this optimization, if we document it :)

Reply:
SGTM, added a code comment in d61cef9